Sources/BraintreePayPal/BTPayPalDriver.m

```diff
@@ -108,6 +112,81 @@ - (void)requestOneTimePayment:(BTPayPalCheckoutRequest *)request
 
 #pragma mark - Helpers
 
+- (void)startPayPalNativeCheckout:(BTPayPalCheckoutRequest *)request
+                    configuration:(BTConfiguration *)configuration
+                        pairingId:(NSString *)payToken
+                       completion:(void (^)(BTPayPalAccountNonce *tokenizedCheckout, NSError *error))completionBlock {
+
+    NSString *payPalClientID = [configuration.json[@"paypal"][@"clientId"] asString];
+
+    if (!payPalClientID) {
+      NSError *error = [NSError errorWithDomain:BTPayPalNativeCheckoutErrorDomain
+                                           code:BTPayPalDriverErrorTypeIntegration
+                                       userInfo:@{NSLocalizedDescriptionKey: @"PayPal Native Checkout failed because an empty or nil Client ID was retrieved from the configuration."}];
+      completionBlock(nil, error);
+      return;
+    }
+
+    NSInteger environment;
+    if ([configuration.environment isEqualToString: PayPalEnvironmentProduction]) {
+      environment = PPCEnvironmentLive;
+    }
+    else if ([configuration.environment isEqualToString: PayPalEnvironmentSandbox]) {
+      environment = PPCEnvironmentSandbox;
+    } else {
+      NSError *error = [NSError errorWithDomain:BTPayPalNativeCheckoutErrorDomain
+                                           code:BTPayPalDriverErrorTypeIntegration
+                                       userInfo:@{NSLocalizedDescriptionKey: @"PayPal Native Checkout failed because an invalid environment identifier was retrieved from the configuration."}];
+      completionBlock(nil, error);
+      return;
+    }
+
+    if (![BTAppContextSwitcher sharedInstance].payPalReturnURL) {
+      NSError *error = [NSError errorWithDomain:BTPayPalNativeCheckoutErrorDomain
+                                           code:BTPayPalDriverErrorTypeIntegration
+                                       userInfo:@{NSLocalizedDescriptionKey: @"PayPal Native Checkout failed because no payPalReturnURL was found on the BTAppContextSwitcher singleton"}];
```

Comment:
We could make this text more actionable for merchants:

```suggestion
                                       userInfo:@{NSLocalizedDescriptionKey: @"PayPal Native Checkout failed because the payPalReturnURL on BTAppContextSwitcher is nil. This value cannot be nil."}];
```